common/djangoapps/util/help_desk.py

```diff
@@ -0,0 +1,443 @@
+        Set `HELPDESK` = 'Freshdesk'
+        in `django.conf.settings`.
+        """
+
+
+    def create_ticket(self, subject, desc, name, email, priority=1, status=2, group_id=None):
+        """
+        Create the given `ticket` in Freshdesk.
+
+        The ticket should have the format specified by the freshdesk api.
+        https://freshdesk.com/api#create_ticket
+        """
+        headers = {'Content-Type': 'application/json'}
+        payload = {
+            'subject': subject,
+            'description': desc,
+            'name': name,
+            'email': email,
+            'priority': priority,
+            'status': status,
+            'group_id': group_id
+        }
+
+        response = requests.post(
+            settings.HELPDESK_URL + '/api/v2/tickets',
+            auth=(settings.HELPDESK_API_KEY, "unused_but_required"),
+            headers=headers,
+            data=json.dumps(payload),
+            allow_redirects=False)
+
+        if response.ok:
+            message = json.loads(response.content)
+            return message['id']
+        else:
+            raise FreshdeskError(response.content, response.status_code, response)
+
+
+    def update_ticket(self, ticket_id, note, tags):
+        """
+        Update the Freshdeck ticket with id `ticket_id` using the given `note`.
+
+        The update should have the format specified by the freshdesk api.
+        https://freshdesk.com/api#add_note_to_a_ticket
+        """
+        headers = {'Content-Type': 'application/json'}
+        payload = {
+            "body" : note,
+            "private" : True
+        }
+
+        response = requests.post(
+            settings.HELPDESK_URL + '/api/v2/tickets/' + str(ticket_id) + '/notes',
+            auth=(settings.HELPDESK_API_KEY, "unused_but_required"),
+            headers=headers,
+            data=json.dumps(payload),
+            allow_redirects=False)
+
+        if response.ok:
+            return json.loads(response.content)
+        else:
+            raise FreshdeskError(response.content, response.status_code, response)
+
+
+    def get_group(self, name):
+        """
+        Find the Freshdesk group named `name`. Groups are cached for
+        CACHE_TIMEOUT seconds.
+
+        If a matching group exists, it is returned as a dictionary
+        with the format specifed by the freshdesk api.
+
+        Otherwise, returns None.
+        """
+        cache = caches['default']
+        cache_key = '{prefix}_group_{name}'.format(prefix=self.CACHE_PREFIX, name=name)
+        cached = cache.get(cache_key)
+        if cached:
+            return cached
+
+        headers = {'Content-Type': 'application/json'}
+        response = requests.get(
+            settings.HELPDESK_URL + '/api/v2/groups',
+            auth=(settings.HELPDESK_API_KEY, "unused_but_required"),
+            headers=headers,
+            allow_redirects=False)
+
+        groups = json.loads(response.content)
```

Comment:
this assignation should be done after the validation and same comment as above

Reply:
This needs to be as it, since groups variable is in use if the response is not 200.